src/compiler/transformers/classFields.ts

```diff
@@ -543,16 +547,36 @@ namespace ts {
 
         function transformClassStaticBlockDeclaration(node: ClassStaticBlockDeclaration, receiver: LeftHandSideExpression) {
             if (shouldTransformPrivateElementsOrClassStaticBlocks) {
-                receiver = visitNode(receiver, visitor, isExpression);
-                const right = factory.createImmediatelyInvokedArrowFunction(visitNode(node.body, visitor, isBlock).statements);
-                const name = createHoistedVariableForClass("_", node);
-                return createPrivateStaticFieldInitializer(
-                    name,
-                    right
-                );
+                const savedClassStaticBlockContainsPrivateFields = classStaticBlockContainsPrivateFields;
+                classStaticBlockContainsPrivateFields = false;
+
+                startLexicalEnvironment();
+                let statements = visitNodes(node.body.statements, visitor, isStatement);
+                statements = factory.mergeLexicalEnvironment(statements, endLexicalEnvironment());
+
+                const result = classStaticBlockContainsPrivateFields ?
+                    transformClassStaticBlockDeclarationWithPrivateFields(statements, node, receiver) :
+                    transformClassStaticBlockDeclarationWithIIFE(statements);

```

Comment:
As far as I can tell the generated variable for the static block when private fields are present goes unused. I would just do this:

```suggestion
                const result = factory.createImmediatelyInvokedArrowFunction(statements);
```

Then we also don't need to track `classStaticBlockContainsPrivateFields`.